sklearn/mixture/_gaussian_mixture_ic.py

```diff
@@ -0,0 +1,744 @@
+"""GaussianMixtureIC"""
+
+# Author: Thomas Athey <tathey1@jhmi.edu>
+# Modified by Benjamin Pedigo <bpedigo@jhu,edu>, Tingshan Liu <tliu68@jhmi.edu>
+
+
+import numpy as np
+from joblib import Parallel, delayed
+import warnings
+
+from . import GaussianMixture
+from ._gaussian_mixture import (
+    _compute_precision_cholesky,
+    _estimate_gaussian_parameters,
+)
+from ..cluster import AgglomerativeClustering
+from ..model_selection import ParameterGrid
+from ..utils import check_scalar, check_array
+from ..utils.validation import check_is_fitted
+from ..exceptions import ConvergenceWarning
+from ..base import BaseEstimator, ClusterMixin
+
+
+class GaussianMixtureIC(ClusterMixin, BaseEstimator):
+    """
```

Comment:
@tliu68 I think this looks good! The only main feedback I have is to try and look through `sklearn` and find similar parameter uses, and choose the names based on that, since open source packages like to name things based on precedent. I think `LassoLarsIC` and some of the spectral clustering methods might be especially useful. 

For an example, here you use `selection_criteria`, while `LassoLarsIC` uses `criterion`.  I know it's minor, but this is something I've come across a lot when PRing.